src/sync_template.sh

```diff
@@ -253,19 +276,13 @@ function pull_source_changes() {
   info "finished pulling from the source."
   info "logging out from source ${SOURCE_REPO_HOSTNAME}."
 
-  if [[ -n "${SRC_SSH_PRIVATEKEY_ABS_PATH}" ]] &>/dev/null; then
-    info "we are using ssh for the source repo. No need to logout."
-  elif [[ -n "${SOURCE_GH_TOKEN}" ]] &>/dev/null; then    
-    gh auth switch      
-  fi
-
   if [ "$pull_has_issues" == true ] ; then
     warn "There had been some git pull issues."
     warn "Maybe a merge issue."
     warn "We go on but it is likely that you need to fix merge issues within the created PR."
   fi
 
-
+  gh_login_target_github "${GITHUB_SERVER_URL}"
```

Comment:
Remark:
out of a logic algorithm sight, need to be checked if e.g. the delete file function is done afterwards.
- [ ] @AndreasAugustin check the algorithm related login/logout